var/spack/repos/builtin/packages/xsd/package.py

```diff
@@ -21,6 +21,8 @@ class Xsd(MakefilePackage):
     depends_on('xerces-c')
     depends_on('libtool', type='build')
 
+    patch('fix_include_iostream.patch')
```

Comment:
Can you add a simple comment on top of the directive, like:
```suggestion
    # Adds a missing include for <iostream> in a source file
    patch('fix_include_iostream.patch')
```
?

Reply:
I had to adapt the patch from [here](https://git.codesynthesis.com/cgit/libxsd-frontend/libxsd-frontend/commit/?id=5029f8665190879285787a9dcdaf5f997cadd2e2), the original version has paths expressed relative to a subdirectory of the source (when the source is the tarball, which apparently contains what corresponds to multiple git repositories). If you can tell me how to specify both the URL of the patch and the fact that it should be applied from a subdirectory, I'm happy to update the PR.

Reply:
I think using this:
```suggestion
    patch(
        'https://git.codesynthesis.com/cgit/libxsd-frontend/libxsd-frontend/patch/?id=5029f8665190879285787a9dcdaf5f997cadd2e2',
        sha256='d57e0aed8784d2b947983209b6513c81ac593c9936c3d7b809b4cd60d4c28607',
        working_dir='libxsd-frontend'
    )
```
and deleting the patch should work (or at least it works locally for me).